specifyweb/frontend/js_src/lib/components/DataModel/businessRuleDefs.ts

```diff
@@ -207,17 +212,47 @@ export const businessRuleDefs: MappedBusinessRuleDefs = {
             }
           );
         }
+        // Flag as invalid if no determinations are checked as current
         if (
           determination.collection !== undefined &&
           !determination.collection.models.some(
             (c: SpecifyResource<Determination>) => c.get('isCurrent')
           )
         ) {
-          determination.set('isCurrent', true);
+          return {
+            isValid: false,
+            reason: resourcesText.currentDeterminationRequired(),
+            saveBlockerKey: CURRENT_DETERMINATION_KEY,
+            resource: determination.collection.related,
+          };
         }
-        return { isValid: true };
+        return {
+          isValid: true,
+          saveBlockerKey: CURRENT_DETERMINATION_KEY,
+          resource: determination.collection?.related,
+        };
```

Comment:
(just a note) 
While this is functional for now, some refactoring in the future may be required. 
Currently, there is not a way to handle if there are multiple blockers which need to be updated with a single field update.
We _could_ utilize `setSaveBlockers` directly, but there may be certain logic in `businessRules` which would be important to execute. And preferably, the `setSaveBlocker` functionality and thinking should be as separate as possible from BusinessRules (i.e., only worry about the BusinessRuleResult and let the internal logic handle setting the saveBlockers). 

Eventually, business rules should be refactored to be able to handle accommodating the case.

Reply:
An additional refactoring note: the other callbacks for business rules (`customInit`, `onRemoved`, and `onAdded`): 

https://github.com/specify/specify7/blob/4e5722aabb4ae3e8afd2ee487f4bea68eaff83d1/specifyweb/frontend/js_src/lib/components/DataModel/businessRuleDefs.ts#L27-L35

should be refactored to have the same return type as the `fieldChecks` callback for the same reasons as stated in the discussion comment above